src/RevProc.cc

```diff
@@ -1905,36 +1885,55 @@ bool RevProc::ClockTick( SST::Cycle_t currentCycle ){
         DependencyClear(HartID, &(Pipeline.front().second));
       }
       Pipeline.erase(Pipeline.begin());
-      GetRegFile(HartID)->SetCost(0);
+      RegFile->SetCost(0);
     }else{
       // could not retire the instruction, bump the cost
       Pipeline.front().second.cost++;
     }
   }
-  if( GetPC() == 0x00ull ){
+  // Check for completion states and new tasks
+  // @DAVE: I feel like the below two blocks of code have some redundancy but its working and
+  //        if I comment out the first block (HartToDecode based checks) the AMO tests fail
+  //        so there must be something I'm not understanding about the two. If there isn't redundancy
+  //        great, I'll uncomment this little letter I left you
+  if( RegFile->GetPC() == 0x00ull ){
```

Comment:
`0` can always be compared against another type and will be automatically converted. It is never lossy. So something `0x000000000llu` (extreme example) is unnecessary and can be simply `0`.

Reply:
This is how it's been used throughout RevProc... Up to @jleidel . Not sure if there was a reason for doing it that way originally

Reply:
Being the minimalist I am, I rarely compare equal or non-equal to 0 -- I would use:
```c++
if( !RegFile->GetPC() ){
```